Casks/b/bbedit@14.rb

```diff
@@ -10,7 +10,10 @@
 
   livecheck do
     url "https://versioncheck.barebones.com/BBEdit.xml"
-    regex(/BBEdit[._-]v?(#{version.major}(?:\.\d+)+)\.dmg/i)
+    strategy :xml do |xml|
+      xml.get_elements("//key[text()='SUFeedEntryShortVersionString']").map { |item| item.next_element&.text&.strip }
+         .select { |item| item.start_with?(version.major) }
```

Comment:
```suggestion
    regex(/^v?(#{version.major}(?:\.\d+)*)$/i)
    strategy :xml do |xml, regex|
      xml.get_elements("//key[text()='SUFeedEntryShortVersionString']").map { |item| item.next_element&.text&.strip }
         .select { |item| item.match?(regex) }
```

Checking for a string that starts with `14` would also match something like 140.2.3. That's not likely to happen but there's something to be said for accurately implementing our intention (only matching 14.x versions). The existing regex matches versions like `14`, `14.1`, `14.1.2`, etc. and we should maintain that behavior.

We could check for versions that start with `14.` but we wouldn't be able to match a major version without more than one version part (e.g., `14` vs. `14.1`). That's not a concern at this point (upstream is at 14.6.9 and and they may never publish versions that only have one part anyway) but we can easily maintain the aforementioned behavior by using a regex with `item.match?(regex)` as the `#select` criteria.

Reply:
It seems like the style audit wants us to use `#grep` here.